zetaclient/chains/evm/signer/signer.go

```diff
@@ -241,74 +246,57 @@ func (signer *Signer) SignRevertTx(txData *OutboundData) (*ethtypes.Transaction,
 	if err != nil {
-		return nil, fmt.Errorf("pack error: %w", err)
+		return nil, fmt.Errorf("onRevert pack error: %w", err)
 	}
 
 	tx, _, _, err := signer.Sign(data,
 		signer.zetaConnectorAddress,
+		big.NewInt(zeroValue),
 		txData.gasLimit,
 		txData.gasPrice,
 		txData.nonce,
 		txData.height)
 	if err != nil {
-		return nil, fmt.Errorf("Sign error: %w", err)
+		return nil, fmt.Errorf("sign onRevert error: %w", err)
 	}
 
 	return tx, nil
 }
 
 // SignCancelTx signs a transaction from TSS address to itself with a zero amount in order to increment the nonce
-func (signer *Signer) SignCancelTx(nonce uint64, gasPrice *big.Int, height uint64) (*ethtypes.Transaction, error) {
-	// TODO: use EIP-1559 transaction type
-	// https://github.com/zeta-chain/node/issues/1952
-	tx := ethtypes.NewTransaction(nonce, signer.tssSigner.EVMAddress(), big.NewInt(0), 21000, gasPrice, nil)
-
-	hashBytes := signer.ethSigner.Hash(tx).Bytes()
-	sig, err := signer.tssSigner.Sign(hashBytes, height, nonce, signer.chain, "")
-	if err != nil {
-		return nil, err
-	}
-
-	pubk, err := crypto.SigToPub(hashBytes, sig[:])
-	if err != nil {
-		signer.logger.Std.Error().Err(err).Msgf("SigToPub error")
-	}
-
-	addr := crypto.PubkeyToAddress(*pubk)
-	signer.logger.Std.Info().Msgf("Sign: Ecrecovery of signature: %s", addr.Hex())
-	signedTX, err := tx.WithSignature(signer.ethSigner, sig[:])
+func (signer *Signer) SignCancelTx(txData *OutboundData) (*ethtypes.Transaction, error) {
+	tx, _, _, err := signer.Sign(
+		nil,
+		signer.tssSigner.EVMAddress(),
+		big.NewInt(zeroValue), // zero out the amount to cancel the tx
+		evm.EthTransferGasLimit,
+		txData.gasPrice,
+		txData.nonce,
+		txData.height,
+	)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("SignCancelTx error: %w", err)
 	}
 
-	return signedTX, nil
+	return tx, nil
 }
 
 // SignWithdrawTx signs a withdrawal transaction sent from the TSS address to the destination
 func (signer *Signer) SignWithdrawTx(txData *OutboundData) (*ethtypes.Transaction, error) {
-	// TODO: use EIP-1559 transaction type
-	// https://github.com/zeta-chain/node/issues/1952
-	tx := ethtypes.NewTransaction(txData.nonce, txData.to, txData.amount, 21000, txData.gasPrice, nil)
-
-	hashBytes := signer.ethSigner.Hash(tx).Bytes()
-	sig, err := signer.tssSigner.Sign(hashBytes, txData.height, txData.nonce, signer.chain, "")
-	if err != nil {
-		return nil, err
-	}
-
-	pubk, err := crypto.SigToPub(hashBytes, sig[:])
-	if err != nil {
-		signer.logger.Std.Error().Err(err).Msgf("SigToPub error")
-	}
-
-	addr := crypto.PubkeyToAddress(*pubk)
-	signer.logger.Std.Info().Msgf("Sign: Ecrecovery of signature: %s", addr.Hex())
-	signedTX, err := tx.WithSignature(signer.ethSigner, sig[:])
+	tx, _, _, err := signer.Sign(
+		nil,
+		txData.to,
+		txData.amount,
+		evm.EthTransferGasLimit,
+		txData.gasPrice,
+		txData.nonce,
+		txData.height,
+	)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("SignWithdrawTx error: %w", err)
```

Comment:
Not covered by test, sme for other sign error below

Reply:
same commit https://github.com/zeta-chain/node/pull/2306/commits/3db4bef06ed6dee6a142efe5152d1704d9084793